src/test/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -0,0 +1,571 @@
+    private static final Pattern DESC_CLEAN = Pattern.compile("-\\s");
+    private static final Pattern FILE_SEPARATOR_PATTERN =
+            Pattern.compile(Pattern.quote(System.getProperty("file.separator")));
+    private static final Pattern QUOTE_PATTERN = Pattern.compile("\"");
+
+    private Log log;
+    private ModuleDetails moduleDetails;
+    private ScrapeStatus currentStatus;
+    private boolean toScan;
+    private String descriptionText;
+    private DetailNode rootNode;
+    private int propertySectionStartIdx;
+    private int exampleSectionStartIdx;
+    private int parentSectionStartIdx;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+            JavadocTokenTypes.PARAGRAPH,
+            JavadocTokenTypes.LI,
+            JavadocTokenTypes.SINCE_LITERAL,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void beginJavadocTree(DetailNode rootAst) {
+        log = LogFactory.getLog(JavadocMetadataScraper.class);
+        if (isTopLevelClassJavadoc()) {
+            moduleDetails = new ModuleDetails();
+            currentStatus = ScrapeStatus.DESCRIPTION;
+            toScan = false;
+            descriptionText = "";
+            propertySectionStartIdx = -1;
+            exampleSectionStartIdx = -1;
+            parentSectionStartIdx = -1;
+
+            final String filePath = getFileContents().getFileName();
+            String moduleName = getModuleSimpleName();
+            if (moduleName.contains("Check")) {
+                moduleName = moduleName.substring(0, moduleName.indexOf("Check"));
+            }
+            moduleDetails.setName(moduleName);
+            moduleDetails.setFullQualifiedName(getPackageName(filePath));
+            moduleDetails.setModuleType(getModuleType());
+        }
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        if (toScan) {
+            scrapeContent(ast);
+        }
+
+        if (ast.getType() == JavadocTokenTypes.JAVADOC
+            && getParent(getBlockCommentAst()).getType() == TokenTypes.CLASS_DEF) {
+            rootNode = ast;
+            toScan = true;
+        }
+        else if (ast.getType() == JavadocTokenTypes.SINCE_LITERAL) {
+            toScan = false;
+        }
+    }
+
+    @Override
+    public void finishJavadocTree(DetailNode rootAst) {
+        moduleDetails.setDescription(getDescriptionText());
+        if (isTopLevelClassJavadoc()) {
+            try {
+                new XmlMetaWriter().write(moduleDetails);
+            }
+            catch (TransformerException | ParserConfigurationException ex) {
+                log.debug("Exception occured while writing metadata of"
+                                + getModuleSimpleName() + " into XML files: ", ex);
```

Comment:
wrap exception in runtime exception and rethrow it.

Reply:
Errors:
- 'throw' inside 'catch' block ignores the caught exception 
- Prohibited exception 'RuntimeException' thrown

Reply:
@gaurabdg 
1. You can set caught exception as cause in new exception constructor, it is should eliminate error
2. Please choose some other runtime exception or create one.

Reply:
Done